ui/selectors/transactions.js

```diff
@@ -534,3 +539,32 @@ export const submittedPendingTransactionsSelector = createSelector(
       (transaction) => transaction.status === TransactionStatus.submitted,
     ),
 );
+
+const hasUnapprovedTransactionsInCurrentNetwork = (state) => {
+  const { unapprovedTxs } = state.metamask;
+  const unapprovedTxRequests = getApprovalRequestsByType(
+    state,
+    ApprovalType.Transaction,
+  );
+
+  const chainId = getCurrentChainId(state);
+
+  const filteredUnapprovedTxInCurrentNetwork = unapprovedTxRequests.filter(
+    ({ id }) => transactionMatchesNetwork(unapprovedTxs[id], chainId),
+  );
+
+  return filteredUnapprovedTxInCurrentNetwork.length > 0;
+};
+
+export function hasTransactionPendingApprovals(state) {
+  return (
+    hasUnapprovedTransactionsInCurrentNetwork(state) ||
+    [
+      ApprovalType.EthDecrypt,
```

Comment:
Not crucial now but this might be best in a local constant such as `TRANSACTION_APPROVAL_TYPES`.

Reply:
Fixed